jwe/gh402_test.go

```diff
@@ -0,0 +1,87 @@
+
+	decrypt := func(customField bool) {
+		t.Helper()
+		m := jwe.NewMessage()
+		plain, err := jwe.Decrypt([]byte(data), jwa.DIRECT, key, jwe.WithMessage(m))
+		if !assert.NoError(t, err, `jwe.Decrypt should succeed`) {
+			return
+		}
+		if string(plain) != "testing Shamir Secret Sharing" {
+			t.Errorf("expected 'testing Shamir Secret Sharing', got %s", string(plain))
+			return
+		}
+
+		if customField {
+			v, ok := m.ProtectedHeaders().Get("clevis")
+			if !assert.True(t, ok, `m.Get("clevis") should be true`) {
+				return
+			}
+
+			if !assert.IsType(t, Pin{}, v, `result of m.Get("clevis") should be an instance of Pin{}`) {
+				return
+			}
+		}
+	}
+	decrypt(false)
+
+	// register field deserialized and run decryption again
+	jwe.RegisterCustomField("clevis", Pin{})
+	decrypt(true) // now it fails
```

Comment:
I believe this comment is out-of-date now

Reply:
Ah, classic cut-n-paste. :)